flytekit/annotated/executors.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from flyteidl.core import literals_pb2 as _literals_pb2
+
+from flytekit import engine as _flyte_engine
+from flytekit.annotated import context_manager, stuff as _annotated
+from flytekit.common import utils as _utils
+from flytekit.interfaces.data import data_proxy as _data_proxy
+from flytekit.models import literals as _literal_models
+
+
+class BaseExecutor(object):
+    # Should these be class methods?
+    def execute(self, ctx: context_manager.FlyteContext, task: _annotated.PythonTask, inputs_path: str):
```

Comment:
if there is inheritance then why class methods? let me be regular methods so that overriding, is easy and you can have class state